src/main/java/filestorage/controllers/FileController.java

```diff
@@ -24,14 +27,33 @@
     public ResponseEntity<?> listFiles() {
         Set<File> files = fileService.getUserAvailableFiles(getCurrentUser());
 
+        if (files == null) {
+            return notFound();
+        }
+
         return new ResponseEntity<>(files, HttpStatus.OK);
     }
 
-    @RequestMapping(method = RequestMethod.POST)
-    public ResponseEntity<?> createFile(@Valid @RequestBody FileRequest request) {
-        File file = fileService.createFile(request, getCurrentUser());
+    @PostMapping(consumes = "multipart/form-data")
+    public ResponseEntity<?> createFile(MultipartHttpServletRequest request) {
+        try {
+            Iterator<String> iterator = request.getFileNames();
+
+            while (iterator.hasNext()) {
+                String uploadedFile = iterator.next();
+                MultipartFile file = request.getFile(uploadedFile);
+                String filename = file.getOriginalFilename();
+                String content = new String(file.getBytes());
 
-        return new ResponseEntity<>(file, HttpStatus.CREATED);
+                File newFile = new File(filename, content, getCurrentUser());
+
+                fileService.uploadFile(newFile, getCurrentUser());
+            }
+        }
+        catch (Exception ex) {
+            return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
```

Comment:
@Bitluck We should return 400 Bad Request error instead of 500 in case when we can't read a file for some reason. Please, implement returning of 400 error with body like {"error": "File can't be uploaded."}